core/src/main/java/io/grpc/ExponentialBackoffPolicy.java

```diff
@@ -0,0 +1,40 @@
+package io.grpc;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import java.util.Random;
+
+/**
+ * Retry Policy for Transport reconnection.
+ */
+class ExponentialBackoffPolicy {
+  /**
+   * Parameters from https://github.com/grpc/grpc/blob/master/doc/connection-backoff.md
+   */
+  static final double INITIAL_BACKOFF_MILLIS = 1 * 1000;
+  static final double MULTIPLIER = 1.6;
+  static final double MAX_BACKOFF_MILLIS = 2 * 60 * 1000;
+  static final double MIN_CONNECT_TIMEOUT_MILLIS = 20 * 1000;
```

Comment:
This doesn't seem the correct usage of MIN_CONNECT_TIMEOUT. MIN_CONNECT_TIMEOUT is a lower-bound for the amount of time we wait for the connection to succeed, and there isn't any timeout here waiting for a connection to succeed.